docs/001_develop/03_client-capabilities/014_utility-methods/13_uuid-util.mdx

```diff
@@ -0,0 +1,72 @@
+---
+title: 'UUID utilities'
+sidebar_label: 'UUID utilities'
+id: uuid-util
+keywords: [utils, utility, uuid, generation, identifier, unique]
+tags:
+  - utils
+  - utility
+  - uuid
+  - generation
+  - identifier
+  - unique
+---
+
+This provides a robust solution for generating UUIDs within your application, offering both immediate and potentially remote generation methods, with an optional logging feature for development and debugging purposes.
+
+## Key Features
+
+- **UUID Generation:** Leverages the `uuidv4` library to generate UUIDs, ensuring uniqueness and compliance with the UUID v4 standard.
+- **Logging Capability:** Offers an optional parameter to log generated UUIDs, aiding in debugging and tracking ID generation.
+- **Configurable Service:** While the current implementation directly uses `uuidv4`, future enhancements may include customizable configurations for UUID generation.
+
+## Use Cases
+
+This module is designed for:
+- Generating a Universally Unique Identifier
+
+## Interfaces and Services
+
+- **`UUIDConfig` Interface:** Intended for future use to customize UUID generation through configuration options.
+- **`UUID` Interface:** Defines the contract for UUID generation services, including methods for creating IDs and remote IDs, with optional logging.
+- **`UUIDService` Class:** Implements the `UUID` interface, providing default behavior for UUID generation and logging.
+
+## Examples
+
+#### Generating a UUID
+
+```typescript
+import { UUID } from '@genesislcap/foundation-utils';
+import { FASTElement } from '@microsoft/fast-element';
+
```

Comment:
We should be using GenesisElement instead of FASTElement from web-core package. If ypu could please make the same change in below example code as well.